rest/utilities_testing.go

```diff
@@ -1386,6 +1369,41 @@ func (bt *BlipTester) SendRev(docId, docRev string, body []byte, properties blip
 
 }
 
+func AdminChannelGrant(collection *db.DatabaseCollection, adminChannels string) (collectionAdminChannels string) {
+	collectionName := collection.Name()
+	scopeName := collection.ScopeName()
+	if base.IsDefaultCollection(scopeName, collectionName) {
+		return adminChannels
+	}
+
+	return fmt.Sprintf(`"collection_access":{%q: {%q: {%s}}}`, scopeName, collectionName, adminChannels)
+}
+
+func (bt *BlipTester) BlipCollectionSetup(collection *db.DatabaseCollection) (blip.Properties, error) {
```

Comment:
This appears to be pretty much the same as `BlipTesterClient.BlipClientCollectionSetup` so I'm wondering if we can share the code

Reply:
I agree I'd much prefer that. Having trouble getting my head around how to approach that with passing BlipTester or BliptesterClient type into a singular function/method? 
Could declare the function to take both types as parameters (`BlipTesterCollectionSetup(*BlipTester, *BlipTesterClient, collection)` ) then just nil BlipTester or BlipClient respectively when using it in different tests? As far as I know there is no way to get BlipTester type from BlipTesterClient or vice versa?

Reply:
There's two `BlipTesters` inside a `BlipTesterClient` which you might be able to use (`BlipTesterClient.pushReplication.bt`) - you're already chosing `pushReplication` inside the other method to send the getCollections request so in both cases I think you have a `blip.Sender` type.